### PR TITLE
Introduce BgCheckFlags Defines

### PR DESCRIPTION
**Pull request by engineer124**
This PR introduces defines for all the different BgCheckFlags. The only ones I'm unsure of are the 0x400/0x800/0x1000 which are newly introduced in MM that aren't in OoT

### CODE REVIEW

**Review by hensldm — APPROVED** [review]
LGTM

File: include/z64actor.h [hunk 1]
```diff
@@ -132,6 +132,20 @@ typedef struct {
...
 } ActorShape; // size = 0x30
 
+#define BGCHECKFLAG_GROUND (1 << 0) // Standing on the ground
+#define BGCHECKFLAG_GROUND_TOUCH (1 << 1) // Has touched the ground (only active for 1 frame)
+#define BGCHECKFLAG_GROUND_LEAVE (1 << 2) // Has left the ground (only active for 1 frame)
+#define BGCHECKFLAG_WALL (1 << 3) // Touching a wall
+#define BGCHECKFLAG_CEILING (1 << 4) // Touching a ceiling
+#define BGCHECKFLAG_WATER (1 << 5) // In water
+#define BGCHECKFLAG_WATER_TOUCH (1 << 6) // Has touched water (reset when leaving water)
+#define BGCHECKFLAG_GROUND_STRICT (1 << 7) // Strictly on ground (BGCHECKFLAG_GROUND has some leeway)
```
Comment (tom-overton) [inline include/z64actor.h:142]: Can the comment here (or for `BGCHECKFLAG_GROUND`) elaborate on what this "leeway" is? I see in the removed docs that this flag is `Similar to & 0x1 but with no velocity check and is cleared every frame`. Is that the difference?

Comment (engineer124) [reply include/z64actor.h:142]: I made a pastebin with rough docs of the relevant function:
https://pastebin.com/Mc2c9m3N

There was slightly more distinction in OoT due to the slightly above ground check (`< 11.0f`) being in `Actor_BgCheckLeaveGround`, which gave leeway to `BGCHECKFLAG_GROUND` being set even when the actor was single unit digits above the floor, while `BGCHECKFLAG_GROUND_STRICT` couldn't have that gap.

But in MM, it's been moved to the main actor bgcheck ground function, meaning `BGCHECKFLAG_GROUND` and `BGCHECKFLAG_GROUND_STRICT` seem to be more in sync with one another.

I think the original comment `Similar to & 0x1 but with no velocity check and is cleared every frame` is better here and will revert.

**Review by tom-overton — APPROVED** [review]

File: include/z64actor.h [hunk 2]
```diff
@@ -132,6 +132,20 @@ typedef struct {
...
+#define BGCHECKFLAG_WALL (1 << 3) // Touching a wall
+#define BGCHECKFLAG_CEILING (1 << 4) // Touching a ceiling
+#define BGCHECKFLAG_WATER (1 << 5) // In water
+#define BGCHECKFLAG_WATER_TOUCH (1 << 6) // Has touched water (reset when leaving water)
+#define BGCHECKFLAG_GROUND_STRICT (1 << 7) // Similar to BGCHECKFLAG_GROUND but with no velocity check and is cleared every frame
+#define BGCHECKFLAG_CRUSHED (1 << 8) // Crushed between a floor and ceiling (triggers a void for player)
+#define BGCHECKFLAG_PLAYER_WALL_INTERACT (1 << 9) // Only set/used by player, related to interacting with walls
+#define BGCHECKFLAG_PLAYER_400 (1 << 10) // 
+#define BGCHECKFLAG_PLAYER_800 (1 << 11) // 
+#define BGCHECKFLAG_PLAYER_1000 (1 << 12) //
```
Comment (AngheloAlf) [inline include/z64actor.h:147]: If only we had insight from player :despair:

**Review by AngheloAlf — APPROVED** [review]

**Pull request merged**